apps/wing/src/commands/test.ts

```diff
@@ -254,10 +271,18 @@ async function testSimulator(synthDir: string, options: TestOptions) {
   const testRunner = s.getResource("root/cloud.TestRunner") as std.ITestRunnerClient;
   const tests = await testRunner.listTests();
   const filteredTests = pickOneTestPerEnvironment(tests);
+
+  let testsToRun;
   const results = new Array<std.TestResult>();
 
+  if (options.testFilter) {
+    testsToRun = filterTestsByRegex(filteredTests, options.testFilter);
+  } else {
+    testsToRun = filteredTests;
+  }
+
   // TODO: run these tests in parallel
-  for (const path of filteredTests) {
+  for (const path of testsToRun) {
```

Comment:
it filters only the tests of the simulator, can you please put all the logic in `filterTestsByRegex`, and then call it in each of the targets? 
another option will be putting the logic in`testRunner.listTests()` or `pickOneTestPerEnvironment` (please pick the option that will require you to duplicate the least amount of code),
also, I suspect we randomize test names- can you please verify that?